modules/tictactoe.js

```diff
@@ -0,0 +1,250 @@
+        if (game.thread !== thread) {
+            continue;
+        }
+
+        //If there is already a game with this player, return it
+        if (game.noughts === player || game.crosses === player) {
+            return game;
+        }
+    } 
+}
+
+/**
+ * Places a piece in a certain game
+*/
+function place(game, player, move) {
+    //Work out what piece we are using.
+    var piece = "X",
+        board = game.board;
+
+    if (player === game.noughts) {
+        piece = "O";
+    }
+
+    //It's not even your turn
+    if (player !== game.turn) {
+        return "Invalid Move! (it's not your turn " + player + ")";
+    }
+
+    //You tried to put it on top of an existing piece
+    if (board[move[0]][move[1]] !== '-') {
+        return "Invalid Move! (there's already something there " + player + ")";
+    }
+
+    //Make your move
+    board[move[0]][move[1]] = piece;
+
+    //Swap the current player
+    if (player === game.noughts) {
+        game.turn = game.crosses;
+    } else {
+        game.turn = game.noughts;
+    }
+}
+
+/**
+ * Creates a new TicTacToe game
+*/
+function createGame(player1, player2, thread) {
+    var board = [['-','-','-'],['-','-','-'],['-','-','-']],
+        noughts = player1,
+        crosses = player2,
+        game = {
+            thread: thread,
+            board: board,
+            noughts: player1,
+            crosses: player2,
+            turn: crosses
+        };
+
+    return game;
+}
+
+/**
+*prints the game to the chat
+*/
+function printGame(game) {
+    var output = "",
+        board = game.board;
+
+    for (var i = 0; i < board.length; ++i) {
+        for (var j = 0; j < board[i].length; ++j) {
+            if (j > 0) {
+                output += "|";
+            }
+
+            output += board[i][j];
+        }
+
+        if (i < 2) {
+            output += "\n";
+        }
+    }
+
+    return output;
+}
+
+/**
+* Checks to see if a player has won. Returns the winner or undefined
+*/
+function checkWin(game, player) {
+    var board = game.board,
+        piece = player === game.nought ? "O" : "X";
+
+    //Check rows
+    for (var i in board) {
+        var row = board[i];
+
+        if (row[0] === piece && row[1] === piece && row[2] === piece) {
+            return player;
+        }
+    }   
+
+    //Check columns
+    for (var i = 0; i < 3; ++i) {
+        if (board[0][i] === piece && board[1][i] === piece && board[2][i] === piece) {
+            return player;
+        }
+    }
+
+    //Check Diagonal
+    if (board[0][0] === piece && board[1][1] === piece && board[2][2] === piece) {
+        return player;
+    }
+
+    //Check Anti-Diagonal
+    if (board[0][2] === piece && board[1][1] === piece && board[2][0] === piece) {
+        return player;
+    }
+}
+
+exports.match = function(text) {
+    return text.startsWith(this.platform.commandPrefix + 'tictactoe')
+        || text.startsWith(this.platform.commandPrefix + 'move')
```

Comment:
rather than having to type move you could force it to match when a person types. there are 2 other arguments passed to the match function